scripts/trace_docs_deps.py

```diff
@@ -0,0 +1,104 @@
+"""
```

Comment:
I'm really not a big fan of creating an entirely separate `scripts` directory for this but I didn't want to place it in `docs` since Docusaurus renders everything in there. Thoughts on where to keep this? Tests will have be adjusted accordingly.

Reply:
A scripts directory is not necessarily a bad thing for CI/CD, it can help make the pipeline a bit more readable if it calls scripts instead of having all the bash written directly in the pipeline. It does add an additional hop while troubleshooting but it allows you to trace through the pipeline a bit faster as well.

Reply:
Hmm that's a good point! I guess I was reluctant to make a directory in the root of the project but doubt it's a huge deal.

Reply:
Also if the azure pipelines don't have to be defined at the top directory of the repo they could be in their own directory. Then the scripts and all things related to the deployment process could live there as well, possibly under a scripts or bin subfolder.

Reply:
Ah I really like the idea of decluttering the top directory; perhaps not fully in scope for our current sprints but I'll make note of it in a ticket. Thanks!